docs/cookbook/deploy-using-bentoml.md

```diff
@@ -0,0 +1,201 @@
+            "enum": ["sword", "axe", "mace", "spear", "bow", "crossbow"],
+            "type": "string"
+        }
+    }
+}"""
+```
+
+First, we need to define a BentoML service by decorating an ordinary class (`Outlines` here) with `@bentoml.service` decorator. We pass to this decorator some configuration and GPU on which we want this service to run in BentoCloud (here an L4 with 24GB memory):
+
+```python
+import typing as t
+import bentoml
+
+from import_model import BENTO_MODEL_TAG
+
+@bentoml.service(
+    traffic={
+        "timeout": 300,
+    },
+    resources={
+        "gpu": 1,
+        "gpu_type": "nvidia-l4",
+    },
+)
+class Outlines:
+
+    bento_model_ref = bentoml.models.get(BENTO_MODEL_TAG)
+
+    def __init__(self) -> None:
+
+        import outlines
+        import torch
+        self.model = outlines.models.transformers(
+            self.bento_model_ref.path,
+            device="cuda",
+            model_kwargs={"torch_dtype": torch.float16},
+        )
+
+    ...
+```
+
+We then need to define an HTTP endpoint using `@bentoml.api` to decorate the method `generate` of `Outlines` class:
+
+```python
+    ...
+
+    @bentoml.api
+    async def generate(
+        self,
+        prompt: str = "Give me a character description.",
+        json_schema: t.Optional[str] = DEFAULT_SCHEMA,
+    ) -> t.Dict[str, t.Any]:
+
+        import outlines
+
+        generator = outlines.generate.json(self.model, json_schema)
+        character = generator(prompt)
+
+        return character
+```
+
+Here `@bentoml.api` decorator defines `generate` as an HTTP endpoint that accept a JSON request body with 2 fields: `prompt` and `json_schema` (optional, which let http client to provide their own JSON schema). The type hints in the function signature will be used to validate incoming JSON request. You can define as many HTTP endpoints as you want by using `@bentoml.api` to decorate other methods of `Outlines` class.
+
+Now you can save the above codes to `service.py` (or use [this implementation](https://github.com/outlines-dev/outlines/blob/main/examples/bentoml/)). Let's now see how to run the code on the cloud using the BentoML CLI.
+
+## Run locally for testing and debugging
+
+Then you can run a server locally by:
+
+```bash
+bentoml serve .
+```
+
+The server is now active at <http://localhost:3000>. You can interact with it using the Swagger UI or in other different ways:
+
+<details>
+
+<summary>CURL</summary>
+
+```bash
+curl -X 'POST' \
+  'http://rtxbox.jkdf.win:3000/generate' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "prompt": "Give me a character description."
+}'
+```
+
+</details>
+
+<details>
+
+<summary>Python client</summary>
+
+```python
+import bentoml
+
+with bentoml.SyncHTTPClient("http://localhost:3000") as client:
+    response = client.generate(
+        prompt="Give me a character description"
+    )
+    print(response)
+```
+
+</details>
+
+
```

Comment:
Add an example output here to let users know what they can expect.

```suggestion
Expected output:

```bash
{
  "name": "Aura",
  "age": 15,
  "armor": "plate",
  "weapon": "sword",
  "strength": 20
}
```
```